### PR TITLE
Bug lc 291 fix duration

### PR DESCRIPTION
**Pull request by KLV96**
**Jira Ticket:** 
https://cshrdigitalandanalysis.atlassian.net/browse/LC-291

This branch has the implementation for the **c** part of the Jira ticket. Duration is being implemented to face-to-face courses and adds it to the total duration of a course. Duration is being displayed in hours as agreed and presented on the Jira ticket.

### CODE REVIEW

File: src/lib/datetime.ts [hunk 1]
```diff
@@ -36,13 +36,40 @@ function ensureFullICU() {
 	}
 }
 
+export function formatCourseDurationForCourseInlcFaceToFace(d: number) {
```
Comment (kameshganesan) [inline src/lib/datetime.ts]: d = duration?

File: src/lib/datetime.ts [hunk 2]
```diff
@@ -36,13 +36,40 @@ function ensureFullICU() {
...
+	if (!d) {
+		return '-'
+	}
+	let out = ''
+
+	const [hours, hourSeconds] = divmod(d, 3600)
+	if (hours) {
```
Comment (kameshganesan) [inline src/lib/datetime.ts]: Is your IDE indenting with 8 space?. If yes, then that's a lot. Must be 2.

File: src/lib/model.ts [hunk 3]
```diff
@@ -105,8 +105,30 @@ export class Course {
 
 	getDuration() {
 		const durationArray = this.modules.map(m => m.duration)
+		// tslint:disable-next-line:prefer-for-of
+			for (let i = 0; i < this.modules.length; i++) {
```
Comment (kameshganesan) [inline src/lib/model.ts]: use foreach

File: src/lib/model.ts [hunk 4]
```diff
@@ -271,13 +305,18 @@ export class ModuleWithCourse extends Module {
...
-			date = new Date(
+			startDate = new Date(
 				data.dateRanges[0].date + 'T' + data.dateRanges[0].startTime
 			)
+			endDate = new Date(
+				data.dateRanges[0].date + 'T' + data.dateRanges[0].endTime
+			)
 		} else {
-			date = data.date
+			// date = data.date
```
Comment (kameshganesan) [inline src/lib/model.ts]: remove else if not needed.  branch which does nothing as you have commented the only line

File: src/ui/page/search/index.html [hunk 5]
```diff
@@ -30,6 +30,10 @@ <h1 class="heading heading-large heading--flush">
                             </p>
                         {/if}
                         {#each searchResults.combinedResults as result}
+                            {#each result as resultItem}
+                            <p>{resultItem}</p>
```
Comment (kameshganesan) [inline src/ui/page/search/index.html]: indent this p

File: src/lib/datetime.ts (hunk 1 above)
Comment (KLV96) [reply src/lib/datetime.ts:39]: yup, I can't change it as the whole page was built using d as duration

File: src/lib/model.ts [hunk 6]
```diff
@@ -105,8 +105,31 @@ export class Course {
...
+					const durationInSeconds = durationInMinutes * 60
+					// tslint:disable-next-line:indent
+					durationArray[0] = durationInSeconds
+				}
+			}
+		})
+
+		let totalDuration = 0
+		// tslint:disable-next-line:prefer-for-of
+		for (let i = 0; i < durationArray.length; i++) {
```
Comment (kameshganesan) [inline src/lib/model.ts:127]: why not array#reduce, it is meant for this.

**Comment by m24murray** [comment]
@KLV96 - can we have a look at these review comments and rebase with master to resolve the conflicts?

File: src/ui/component/disciteParts/DisciteCourseProperties.html [hunk 7]
```diff
@@ -13,12 +13,12 @@
             <span>Type:</span>
             <span class="lpg-course-type">{$i18n(course.getType() || '' )}</span>
         </li>
-        {#if course.getDuration() && course.getDuration() !== '-' }
+<!--        {#if course.getDuration() && course.getDuration() !== '-' }-->
```
Comment (m24murray) [inline src/ui/component/disciteParts/DisciteCourseProperties.html:16]: Why have we commented this if?

File: src/ui/controllers/booking/cancel.ts [hunk 8]
```diff
@@ -145,7 +144,6 @@ export async function tryCancelBooking(
 		},
 		400: async () => {
 			req.session!.save(() => {
-				req.flash('cancelBookingError', "An error occurred while trying to cancel your booking.")
```
Comment (m24murray) [inline src/ui/controllers/booking/cancel.ts:148]: why is this being removed

File: src/ui/page/booking/choose-date.html [hunk 9]
```diff
@@ -34,12 +34,15 @@ <h2 class="heading-medium u-space-t10 heading--red-top">{ module.title || course
...
+<!--                                <h1>end Date: {event.endDate}</h1>-->
                                 <div class="div-table__row u-clearfix lpg-date-row">
-                                    <div class="div-table__cell lpg-date">{$datetime.formatDate(event.date)}</div>
+                                    <div class="div-table__cell lpg-date">{$datetime.formatDate(event.startDate)}</div>
 
                                     <div class="div-table__cell">
-                                        {$datetime.formatTime(event.date, true)}
-                                        { module.duration ? 'to ' + $datetime.addSeconds(event.date, module.duration, true) : ''}
+                                        {$datetime.formatTime(event.startDate, true)} to {$datetime.formatTime(event.endDate, true)}
+<!--                                        This one needs to be changes the module.duration is wrong as the event.date whould have a start date and an end date-->
```
Comment (m24murray) [inline src/ui/page/booking/choose-date.html:44]: Comments?

File: src/lib/model.ts [hunk 10]
```diff
@@ -316,7 +358,9 @@ export class Event {
 		if (id) {
 			this.id = id!
 		}
-		this.date = date
+		// this.date = date
```
Comment (m24murray) [inline src/lib/model.ts:361]: comments

**Pull request merged**